src/index.rs

```diff
@@ -52,6 +52,18 @@ impl Ord for Point {
     }
 }
 
+impl From<Point<usize>> for Point<isize> {
```

Comment:
Is there any reason we use both `usize` and `isize` instead of just `isize`?

Reply:
Well generally the index should never be negative. And it actually never is negative in the normal screen buffer. So we use `usize` for most stuff internally.

But with the alternate screen buffer, scrolling doesn't actually scroll a buffer, there's not actually anything going up/down, because the alternate screen buffer has only the visible region. As a result the line `0` is always the bottom, so if we want to keep track of everything below that, using negative lines seems reasonable. Thus the `isize` stuff.

It doesn't make much sense to do this for the complete codebase though, since most scenarios never want to keep track of anything outside of the terminal emulator. So for all other places the only thing relevant is the offset from the bottom of all lines, which is `0`. So a `usize` works perfectly.

I've decided to use `event.rs` as a hand-off point for this since it is an intermediate step between `input.rs` and `selection.rs`. So everything talking to it can just use the standard `usize` which are used everywhere and then the selection can take over which actually has a concept of lines below the terminal emulator.

Note that this PR still needs another clean-up pass. So some things might seem nonsensical (haven't looked at the resulting diff yet). But the general idea shouldn't be totally off.